rnnalyse/extractors/base_extractor.py

```diff
@@ -147,38 +145,76 @@ def _print_time_info(self, start_time: float, print_every: int) -> None:
               f'Time: {minutes:>3.0f}m {seconds:>2.2f}s\t'
               f'Speed: {speed:.2f}s/sen')
 
-    def _extract_sentence(self, sentence: Sentence) -> None:
+    def _extract_sentence(self, sentence: LabeledSentence) -> int:
         """ Generates the embeddings of a sentence and writes to file.
 
         Parameters
         ----------
         sentence : Sentence
             The to-be-extracted sentence, represented as a list of strings.
+
+        Returns
+        -------
+        num_extracted: int
+            Number of extracted activations for this sentence.
         """
+
+        sen_activations: PartialActivationDict = self._init_sen_activations(0)
+
```

Comment:
This seems sensible, but it is difficult to be sure if its correct. This would be an excellent functionality to write a test for that mocks the activation reader and tests if the selection actually writes away the correct activations.

@Kaleidophon could you create such a test? In machine you can find several examples of mocking.

Reply:
See above. For testing I would just start opening and issue and collect the function we find especially worthy testing (or, differently put, worthy to prioritize for testing).

Reply:
In general I agree that we should open issues for such things. But at the same time I believe that we are a bit too lacky about this. It is good practice to not accept PRs without some tests that show that they are correct, especially if they are function related (which may be difficult to see from a file). It is a bit strange perhaps to require that in a library that otherwise has no tests, but we have to start from somewhere.

@Kaleidophon  If you find this difficult, I can set something up and ask you to finish the actual implementation of the test logic?

Reply:
I can do that. If you provide some form of template that would be useful to me in order to see what you expect